sycl/test/basic_tests/aspect_usage.cpp

```diff
@@ -0,0 +1,50 @@
+// Test checks Propagate Aspect Usage pass.
+
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager %s -emit-llvm -S -o - | FileCheck %s --check-prefix CHECK-NODES
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager -fno-sycl-early-optimizations %s -emit-llvm -S -o - | FileCheck %s --check-prefix CHECK-NODES
+// RUN: %clangxx -fsycl -flegacy-pass-manager %s -emit-llvm -S -o - | FileCheck %s --check-prefix CHECK-NODES
+// RUN: %clangxx -fsycl -flegacy-pass-manager -fno-sycl-early-optimizations %s -emit-llvm -S -o - | FileCheck %s --check-prefix CHECK-NODES
+
+// fp16 aspect corresponds to constant 5.
+// CHECK-NODES-DAG: !intel_used_aspects ![[NODE:[0-9]+]]
+// CHECK-NODES: ![[NODE]] = !{i32 5}
+
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager %s -emit-llvm -S -o /dev/null 2>&1 | FileCheck %s --check-prefix CHECK-WARNINGS
+// RUN: %clangxx -fsycl -flegacy-pass-manager %s -emit-llvm -S -o /dev/null 2>&1 | FileCheck %s --check-prefix CHECK-WARNINGS
+
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager %s -g -emit-llvm -S -o /dev/null 2>&1 | FileCheck %s --check-prefix CHECK-WARNINGS-DBG -DPATH=%s
+// RUN: %clangxx -fsycl -flegacy-pass-manager %s -g -emit-llvm -S -o /dev/null 2>&1 | FileCheck %s --check-prefix CHECK-WARNINGS-DBG -DPATH=%s
+
+// CHECK-WARNINGS: warning: function '_Z5func1v' uses aspect '5' not listed in `sycl::device_has()`
+// CHECK-WARNINGS-NEXT:  _Z5func1v()
+// CHECK-WARNINGS-NEXT:  _Z5func2v()
```

Comment:
Why do we need those `()` - will they be filled with the actual function prototype if it has arguments?

Reply:
Currently, arguments info is not emitted. Theoretically, it is easy with scalar arguments since I have function's signature in IR. However, I do not have a clear vision how to handle template arguments like `int func(MyStruct<T> &MS)`.

Reply:
In fact, I've realized that I've already added a demangling some time ago but haven't updated test. Now test contains arguments in functions.